psi4/src/psi4/occ/occ_iterations.cc

```diff
@@ -410,5 +451,176 @@ void OCCWave::occ_iterations() {
+            global_dpd_->buf4_init(&R2bb, PSIF_OCC_DPD, 0, ID("[o,o]"), ID("[v,v]"), ID("[o,o]"), ID("[v,v]"), 0,
+                               "RT2_2 <oo|vv>");
+            orbitalDiis->add_entry(16, wogA_vec.get(), wogB_vec.get(), &R1aa, &R1ab, &R1bb, &R2aa, &R2ab, &R2bb,
+                    kappa_bar_[SpinType::Alpha].get(), kappa_bar_[SpinType::Beta].get(), &T1aa, &T1ab, &T1bb, &T2aa, &T2ab, &T2bb);
+            if (orbitalDiis->subspace_size() >= mindiis_) {
+                orbitalDiis->extrapolate(8, kappa_bar_[SpinType::Alpha].get(), kappa_bar_[SpinType::Beta].get(), &T1aa, &T1ab, &T1bb, &T2aa, &T2ab, &T2bb);
+            }
+            global_dpd_->buf4_close(&T1aa);
+            global_dpd_->buf4_close(&T1ab);
+            global_dpd_->buf4_close(&T1bb);
+            global_dpd_->buf4_close(&R1aa);
+            global_dpd_->buf4_close(&R1ab);
+            global_dpd_->buf4_close(&R1bb);
+            global_dpd_->buf4_close(&T2aa);
+            global_dpd_->buf4_close(&T2ab);
+            global_dpd_->buf4_close(&T2bb);
+            global_dpd_->buf4_close(&R2aa);
+            global_dpd_->buf4_close(&R2ab);
+            global_dpd_->buf4_close(&R2bb);
+        } else if (wfn_type_ == "OCEPA") {
+            dpdbuf4 Taa, Tab, Tbb, Raa, Rab, Rbb;
+            global_dpd_->buf4_init(&Taa, PSIF_OCC_DPD, 0, ID("[O,O]"), ID("[V,V]"), ID("[O,O]"), ID("[V,V]"), 0,
+                               "T2 <OO|VV>");
+            global_dpd_->buf4_init(&Raa, PSIF_OCC_DPD, 0, ID("[O,O]"), ID("[V,V]"), ID("[O,O]"), ID("[V,V]"), 0,
+                               "RT2 <OO|VV>");
+            global_dpd_->buf4_init(&Tab, PSIF_OCC_DPD, 0, ID("[O,o]"), ID("[V,v]"), ID("[O,o]"), ID("[V,v]"), 0,
+                               "T2 <Oo|Vv>");
+            global_dpd_->buf4_init(&Rab, PSIF_OCC_DPD, 0, ID("[O,o]"), ID("[V,v]"), ID("[O,o]"), ID("[V,v]"), 0,
+                               "RT2 <Oo|Vv>");
+            global_dpd_->buf4_init(&Tbb, PSIF_OCC_DPD, 0, ID("[o,o]"), ID("[v,v]"), ID("[o,o]"), ID("[v,v]"), 0,
+                               "T2 <oo|vv>");
+            global_dpd_->buf4_init(&Rbb, PSIF_OCC_DPD, 0, ID("[o,o]"), ID("[v,v]"), ID("[o,o]"), ID("[v,v]"), 0,
+                               "RT2 <oo|vv>");
+            orbitalDiis->add_entry(10, wogA_vec.get(), wogB_vec.get(), &Raa, &Rab, &Rbb,
+                    kappa_bar_[SpinType::Alpha].get(), kappa_bar_[SpinType::Beta].get(), &Taa, &Tab, &Tbb);
+            if (orbitalDiis->subspace_size() >= mindiis_) {
+                orbitalDiis->extrapolate(5, kappa_bar_[SpinType::Alpha].get(), kappa_bar_[SpinType::Beta].get(), &Taa, &Tab, &Tbb);
+            }
+            global_dpd_->buf4_close(&Taa);
+            global_dpd_->buf4_close(&Tab);
+            global_dpd_->buf4_close(&Tbb);
+            global_dpd_->buf4_close(&Raa);
+            global_dpd_->buf4_close(&Rab);
+            global_dpd_->buf4_close(&Rbb);
+        }
+    }
+    psio_->close(PSIF_OCC_DPD, 1);
```

Comment:
I have high hopes of some of these dpd closes enabling better occ testing by running one calc after another.